[FLOSS T2] improve storage plugin tutorial by adding sections about ordering and comments (#4178)
ElektraInitiative/libelektra: doc/tutorials/storage-plugins.md

sudo kdb umount user:/tests/storage
```

## Storing Comments

> Most markup languages provide the possibility of adding comments. Elektra can store those comments in its meta-information as well. This can be achieved by setting the meta Key `comment` for the respective configuration Key. The `hosts` plugin currently stores the comments of the file in the respective Elektra configuration Key:

Reviewer: ```suggestion
Most markup languages provide the possibility of adding comments.
Elektra can store those comments in its meta-data as well.
This can be achieved by setting the meta Keys `comment/#` for
the respective configuration Key. Also the `hosts` plugin stores
the comments of the file in the respective Elektra configuration Key:
```

```sh
# Mount empty hosts file

> kdb mount --with-recommends hosts /tests/hosts hosts

Reviewer: ```suggestion
sudo kdb mount --with-recommends hosts /tests/hosts hosts
```

# Add a line to the hosts file containing a comment
echo '127.0.0.1    localhost # test comment' >  `kdb file /tests/hosts`

# Check if the line has been synced successfully
kdb get /tests/hosts/ipv4/localhost
#> 127.0.0.1

kdb meta-ls /tests/hosts/ipv4/localhost
#> comment/#0
#> comment/#0/space
#> comment/#0/start
#> order

kdb meta-get /tests/hosts/ipv4/localhost 'comment/#0'
#>  test comment

# Undo modifications to the key database
kdb rm -r /tests/hosts

> sudo kdb umount /tests/hosts

Reviewer: :+1:

```

## Ordering of Elements

If your plugin also has the ability to store configuration options in a certain order, then this is also support by Elektra. Keys can have the meta Key `order`, which indicates in which order lines should be written back to the configuration file. Inversely, when reading from configuration files, plugins should add the `order` meta Key to the respective KDB entries.

This behavior can be illustrated via the usage of the `hosts` plugin, which honors this convention:

```sh
# Mount empty hosts file

> kdb mount --with-recommends hosts /tests/hosts hosts

Reviewer: ```suggestion
sudo kdb mount --with-recommends hosts /tests/hosts hosts
```
